src/sentry/incidents/endpoints/serializers/workflow_engine_detector.py

```diff
@@ -1,34 +1,325 @@
+
+            try:
+                group_ids = ActionGroupStatus.objects.values_list("group_id", flat=True).filter(
+                    action_id__in=detector_to_action_ids[detector]
+                )
+            except ActionGroupStatus.DoesNotExist:
+                continue
+
+            if group_ids:
+                try:
+                    open_periods = GroupOpenPeriod.objects.filter(
+                        group__in=[group_id for group_id in group_ids]
+                    ).order_by("-date_started")
+                except GroupOpenPeriod.DoesNotExist:
+                    continue
+
+            # TODO: this serializer is half baked
+            if open_periods:
+                serialized_group_open_period = serialize(
+                    open_periods[0], user, WorkflowEngineIncidentSerializer()
+                )
+                result[detector]["latestIncident"] = serialized_group_open_period
+
+    def get_attrs(
+        self, item_list: Sequence[Detector], user: User | RpcUser | AnonymousUser, **kwargs: Any
+    ) -> defaultdict[Detector, dict[str, Any]]:
+        detectors = {item.id: item for item in item_list}
+        result: DefaultDict[Detector, dict[str, Any]] = defaultdict(dict)
+
+        detector_workflow_condition_group_ids = [
+            detector.workflow_condition_group.id
+            for detector in detectors.values()
+            if detector.workflow_condition_group
+        ]
+        detector_trigger_data_conditions = DataCondition.objects.filter(
+            condition_group__in=detector_workflow_condition_group_ids,
+            condition_result__in=[DetectorPriorityLevel.HIGH, DetectorPriorityLevel.MEDIUM],
+        )
+
+        action_filter_data_condition_groups = DataCondition.objects.filter(
+            comparison__in=[
+                detector_trigger.condition_result
+                for detector_trigger in detector_trigger_data_conditions
             ],
-            "dateModified": datetime.now(),
-            "dateCreated": datetime.now(),
-            "createdBy": {},
-            "description": "test",
-            "detectionType": "test",
+        ).exclude(condition_group__in=detector_workflow_condition_group_ids)
+
+        dcgas = DataConditionGroupAction.objects.filter(
+            condition_group__in=[
+                action_filter.condition_group
+                for action_filter in action_filter_data_condition_groups
+            ]
+        ).select_related("action")
+        actions = [dcga.action for dcga in dcgas]
+
+        # add sentry app data
+        organization_id = [detector.project.organization_id for detector in detectors.values()][0]
+        sentry_app_installations_by_sentry_app_id = (
+            self.add_sentry_app_installations_by_sentry_app_id(actions, organization_id)
+        )
+
+        # add trigger and action data
+        serialized_data_conditions: list[dict[str, Any]] = []
+        for trigger in detector_trigger_data_conditions:
```

Comment:
This is a bit of a hack in that I'm building up the data conditions instead of passing the whole list in. If I pass the whole list in I hit a bug that past me thought was ok for some reason https://github.com/getsentry/sentry/blob/master/src/sentry/incidents/endpoints/serializers/workflow_engine_data_condition.py#L76-L78

The only way to get around that though is to refactor the data condition serializer to do the insane mapping I'm doing for incidents starting on line 256, so this seems kinda preferable?